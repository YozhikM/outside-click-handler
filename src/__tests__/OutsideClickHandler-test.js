/* @flow */

import React from 'react';
import { shallow } from 'enzyme';
import OutsideClickHandler from '../OutsideClickHandler';

const ChildrenComponent = () => <span>Children</span>;

describe('OutsideClickHandler', () => {
  it('render()', () => {
    const onOutsideClick = jest.fn();
    const wrapper = shallow(
      <OutsideClickHandler onOutsideClick={onOutsideClick}>
        <ChildrenComponent />
      </OutsideClickHandler>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidMount()', () => {
    const onOutsideClick = jest.fn();
    const wrapper = shallow(
      <OutsideClickHandler onOutsideClick={onOutsideClick}>
        <ChildrenComponent />
      </OutsideClickHandler>
    );
    // $FlowFixMe
    document.addEventListener = jest.fn();
    wrapper.instance().componentDidMount();

    expect(document.addEventListener).toBeCalledWith('click', wrapper.instance().onOutsideClick);
  });

  it('componentWillUnmount()', () => {
    const onOutsideClick = jest.fn();
    const wrapper = shallow(
      <OutsideClickHandler onOutsideClick={onOutsideClick}>
        <ChildrenComponent />
      </OutsideClickHandler>
    );
    // $FlowFixMe
    document.removeEventListener = jest.fn();
    wrapper.instance().componentWillUnmount();

    expect(document.removeEventListener).toBeCalledWith('click', wrapper.instance().onOutsideClick);
  });

  it('onOutsideClick()', () => {
    const e = jest.fn();
    const onOutsideClick = jest.fn();
    const wrapper = shallow(
      <OutsideClickHandler onOutsideClick={onOutsideClick}>
        <ChildrenComponent />
      </OutsideClickHandler>
    );
    wrapper.instance().onOutsideClick(e);

    expect(onOutsideClick).toBeCalled();
  });

  it('setChildNodeRef()', () => {
    const onOutsideClick = jest.fn();
    const wrapper = shallow(
      <OutsideClickHandler onOutsideClick={onOutsideClick}>
        <ChildrenComponent />
      </OutsideClickHandler>
    );
    wrapper.instance().setChildNodeRef(<ChildrenComponent />);

    expect(wrapper.instance().childNode).toEqual(<ChildrenComponent />);
  });
});
