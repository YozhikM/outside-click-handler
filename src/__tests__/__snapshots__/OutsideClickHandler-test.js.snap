// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutsideClickHandler render() 1`] = `
<div>
  <SvgIcon
    file="add"
  />
</div>
`;
